nengo_dl/tests/test_converter.py

```diff
@@ -809,3 +820,182 @@ def relu2(x):
+    )
+
+    with ctx:
+        converter.Converter(
+            model, allow_fallback=False, temporal_model=True, inference_only=True
+        )
+
+
+@pytest.mark.parametrize("Layer", (compat.Lowpass, compat.Alpha))
+def test_lowpass_alpha(Layer, Simulator, rng):
+    pytest.importorskip("keras_spiking")
+
+    inp = x = tf.keras.Input((None, 10))
+    x = tf.keras.layers.Dense(units=10)(x)
+    x = Layer(tau=0.1, trainable=False)(x)
+
+    model = tf.keras.Model(inp, x)
+
+    conv = converter.Converter(model, temporal_model=True, allow_fallback=False)
+
+    inp_vals = rng.uniform(0, 2, size=(8, 50, 10))
+
+    with Simulator(conv.net) as sim:
+        model_out = model.predict(inp_vals)
+        nengo_out = sim.predict(inp_vals)
+
+        # nengo model is off by one timestep due to the synaptic delay
+        assert np.allclose(
+            model_out[:, :-1], nengo_out[conv.outputs[x]][:, 1:], atol=5e-6
+        )
+
+        # check training
+        # TODO: training should work, but I can't find a way to get the training
+        #  behaviour to match due to the different dynamics introduced by the
+        #  one-timestep delay
+        # sim.compile(optimizer=tf.optimizers.SGD(0.1), loss=tf.losses.mse)
+        # sim.fit(
+        #     inp_vals,
+        #     np.concatenate((np.zeros((8, 1, 10)), inp_vals[:, :-1]), axis=1),
+        #     epochs=3,
+        # )
+        #
+        # model.compile(optimizer=tf.optimizers.SGD(0.1), loss=tf.losses.mse)
+        # model.fit(inp_vals, inp_vals, epochs=3)
+        #
+        # model_out = model.predict(inp_vals)
+        # nengo_out = sim.predict(inp_vals)
+        # assert np.allclose(
+        #     model_out[:, :-1], nengo_out[conv.outputs[x]][:, 1:], atol=2e-6
+        # )
+
+    # can't convert trainable layers
+    inp = tf.keras.Input((None, 10))
+    layer = Layer(tau=0.1, trainable=True)
+    model = tf.keras.Model(inp, layer(inp))
+    with pytest.raises(TypeError, match="layer.trainable=False"):
+        converter.Converter(model, temporal_model=True)
+
+    # can't convert layers with non-zero initial level
+    layer.trainable = False
+    tf.keras.backend.set_value(
+        layer.layer.cell.initial_level, np.ones(layer.layer.cell.initial_level.shape)
+    )
+    model = tf.keras.Model(inp, layer(inp))
+    with pytest.raises(TypeError, match="initial_level != 0"):
+        converter.Converter(model, temporal_model=True)
+
+    # can't convert layers with individual taus
+    tf.keras.backend.set_value(
+        layer.layer.cell.initial_level, np.zeros(layer.layer.cell.initial_level.shape)
+    )
+    tf.keras.backend.set_value(
+        layer.layer.cell.tau_var, np.arange(10).reshape(layer.layer.cell.tau_var.shape)
+    )
+    model = tf.keras.Model(inp, layer(inp))
+    with pytest.raises(TypeError, match="different tau values"):
+        converter.Converter(model, temporal_model=True)
+
+
+def test_time_distributed():
+    inp = x = tf.keras.Input((10, 8))
```

Comment:
How much work would it be to also support `None` for the number of timesteps?

Reply:
Never mind, I realized this (seems to) work, I was just getting an exception in the second part of this test when I switched it.